crowbar_framework/spec/lib/crowbar/upgrade_status_spec.rb

```diff
@@ -77,47 +77,51 @@ def new_status
       expect(subject.current_step).to eql :upgrade_prechecks
       other_status = new_status
       expect(other_status.running?).to be false
-      expect(subject.start_step).to be true
+      expect(subject.start_step(:upgrade_prechecks)).to be true
       other_status.load
       expect(other_status.running?).to be true
     end
 
     it "determines whether a given step is running" do
       expect(subject.current_step).to eql :upgrade_prechecks
       expect(subject.running?(:upgrade_prepare)).to be false
-      expect(subject.start_step).to be true
+      expect(subject.start_step(:upgrade_prechecks)).to be true
       expect(subject.running?(:upgrade_prepare)).to be false
     end
 
     it "determines whether current step is running" do
       expect(subject.current_step).to eql :upgrade_prechecks
       expect(subject.current_step_state[:status]).to eql :pending
       expect(subject.running?).to be false
-      expect(subject.start_step).to be true
+      expect(subject.start_step(:upgrade_prechecks)).to be true
       expect(subject.running?).to be true
       expect(subject.running?(:upgrade_prepare)).to be false
     end
 
     it "moves to next step when requested" do
       expect(subject.current_step).to eql :upgrade_prechecks
       expect(subject.current_step_state[:status]).to eql :pending
-      expect(subject.start_step).to be true
+      expect(subject.start_step(:upgrade_prechecks)).to be true
       expect(subject.end_step).to be true
       expect(subject.current_step).to eql :upgrade_prepare
     end
 
     it "does not move to next step when current one failed" do
       expect(subject.current_step).to eql :upgrade_prechecks
-      expect(subject.start_step).to be true
-      expect(subject.end_step(false, failure: "error message")).to be false
+      expect(subject.start_step(:upgrade_prechecks)).to be true
+      expect(
+        subject.end_step(
+          :upgrade_prechecks, false, false, failure: "error message"
```

Comment:
Code smell - this `false, false` is unintuitive which to me hints that the design is not quite right.  I gave more details in my review, but just wanted to add a mention of the useful maxim from BDD which I found really helpful to remember: tests are a good opportunity to ["write the code you wish you had"](http://waysysweb.com/unittestframework/testing_as_design.htm).